core/interfaces.go

```diff
@@ -135,3 +137,12 @@ type CertificateAuthorityDatabase interface {
 	IncrementAndGetSerial(*gorp.Transaction) (int64, error)
 	Begin() (*gorp.Transaction, error)
 }
+
+type DNSResolver interface {
```

Comment:
I think we need to put a comment on this interface. Perhaps:
// DNSResolver defines the functionality common to DNS resolution services and batman.